src/practitest_firecracker/practitest.clj

```diff
@@ -18,16 +22,16 @@
 (defn build-uri [base-uri resource-uri-template & params]
   (apply format (str base-uri resource-uri-template) params))
 
-(defn api-call [{:keys [credentials uri method query-params form-params]}]
+(defn api-call [{:keys [credentials uri method query-params form-params max-api-rate]}]
   (assert (not (and query-params form-params))
           "both `query-params` and `form-params` can't be specified")
   (loop [results []
          uri     uri
-         params  (cond-> {:basic-auth credentials
-                          :throw-exceptions false
-                          :as :json}
-                   query-params (assoc :query-params query-params)
-                   form-params  (assoc :form-params form-params :content-type :json))]
+         params  (cond-> {:basic-auth          credentials
+                          :throw-exceptions    false
+                          :as                  :json}
+                   query-params (assoc :query-params (conj query-params {:source "firecracker" :firecracker-version fc-version}) :socket-timeout max-api-rate :connection-timeout max-api-rate)
+                   form-params  (assoc :form-params (conj form-params {:source "firecracker" :firecracker-version fc-version}) :content-type :json :socket-timeout max-api-rate :connection-timeout max-api-rate))]
```

Comment:
This is not how API rate limit works. You should implement throttling. If the rate limit is 30 (default), it means that firecracker will do at most 30 calls per minute. If it's 100, it will do at most 100 calls per minute.
`socket-timeout` just means how long to wait before throwing timeout exception if you don't receive answer.
`connection-timeout` just means how long to wait when connecting to the api server before throwing timeout exception.
None of it controls how many calls per minute firecracker does.